core/roboconf-agent/src/main/java/net/roboconf/agent/internal/misc/AgentUtils.java

```diff
@@ -112,6 +118,39 @@ public static void copyInstanceResources( Instance instance, Map<String,byte[]>
 
 
 	/**
+	 * Executes a script resource on a given instance.
+	 * @param instance an instance
+	 * @throws InterruptedException
+	 * @throws IOException
+	 */
+	public static List<ExecutionResult> executeScriptResources( Instance instance, String applicationName, String scopedInstanceName ) throws IOException, InterruptedException {
+
+		File dir = InstanceHelpers.findInstanceDirectoryOnAgent( instance );
+		Utils.createDirectory(dir);
```

Comment:
You should not create anything here.
Either the scripts exists and you execute it, or you do nothing.